### DISCUSSION TITLE
Preview feature feedback: `nativeDistinct`

### DISCUSSION THREAD

**Discussion by janpio** [prisma/prisma#22287] [Feedback]
Support for the **database native `DISTINCT` queries** with [our `distinct` API](https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#select-distinct) is in preview behind the `nativeDistinct` preview feature. 

Please share your feedback about `nativeDistinct` released in [v5.7.0](https://github.com/prisma/prisma/releases/tag/5.7.0) in this discussion.

- If you encounter a bug, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repository.
- If the feature is working well for you, please share this in a comment below or leave a 👍 on this discussion.
- Need help or have questions? Join the [Prisma Discord](https://pris.ly/discord) community and ask away! We're here to support you.

**GA planned for September 2025 - February 2026**

**Comment by meotimdihia** [comment]
`Native DISTINCT` doesn't work when we used `orderBy`
```typescript
          await fastify.prisma.user_info.findMany({
            distinct: ["role"],

            select: {
              role: true
            },
            orderBy: {
              role: {
                sort: "desc",
                nulls: "last"
              }
            }
          })
```

or    use `take` and `limit`
```typescript
          await fastify.prisma.user_info.findMany({
            distinct: ["role"],
            take: req.query.limit,
            skip: req.query.page * req.query.limit,
            select: {
              role: true
            }
          })
```

I created the issue: https://github.com/prisma/prisma/issues/22734

It looks like we have to work around this problem by using subqueries: https://stackoverflow.com/questions/9795660/postgresql-distinct-on-with-different-order-by

**Comment by KilianB** [comment]
Has this feature request been dropped? Druue unassigned herself and there hasn't been made any progress for almost a year. Native distincts are really a pitfall in certain situations as performance penalty for in memory usage can be a few thousand percent.

**Comment by wmadden** [comment]
A quick update, we plan to release this preview feature to GA between September 2025 - February 2026. Read more here: https://github.com/prisma/prisma/discussions/26136

**Comment by UnderKoen** [comment]
Been using this, and has decreased the load on our servers a lot! Can we create a optimisation for when `distinct` and `select` are the same? `select distinct type ...` is a lot more effienct than `select distinct on (type) id, type`
